### ISSUE TITLE
Queries are very slow even in a basic table when the previous query was too slow

### ISSUE THREAD

**Issue by erdusa** [OmniDB/OmniDB#946]
When a query was very slow in its execution, the following queries also became slow regardless of whether the table consulted is small, in this case, the solution is to close omniDB and open it again.
OmniDB 2.15.0
PostgreSQL 10.5

**Comment by wind39** [comment]
Hi @erdusa ,

Nice to hear from you again :)

Please provide more information so we can try to reproduce this. Suppose you are talking about:

- Q1: a real slow query which was executed first and was also slow on OmniDB. So far so good.
- Q2: a fast query which was executed after Q1 and was slow on OmniDB. But executing elsewhere, the query is fine.

Please answer this, providing as many details as you can:

- Were Q1 and Q2 executed in the same Query Tab?
- When you executed Q1, did you let it run to the end, or did you cancel it by clicking on the Cancel button?
- When you executed Q1, was autocommit enabled or disabled?
- Did you change autocommit after executing Q1 and before executing Q2?
- Does Q2 perform fast when you try to execute it in a different, new Query Tab, or is it always slow after execution of Q1?
- If Q1 is not executed at all, does Q2 perform fast as it should?
- Are you using OmniDB server or app?
- If you are using OmniDB server, when you say "solution is to close OmniDB and open it again", do you have to restart server as well or only the browser?
- Does this issue happen in Console Tab?

And also other information that you may find relevant, please.

Thanks in advance,
William

**Comment by erdusa** [comment]
Yes, that is the case, Q1 is fast, Q2 is slow, when I execute Q1 after Q2., now Q1 is slow, but at the finish OmniBD shows the duration is short though it waits for a long time for finish and show results.
This has happened to me several times, when I have this behavior again, I will answer your questions, I hope that is soon.

**Comment by erdusa** [comment]
Hi @wind39, look this:
![image](https://user-images.githubusercontent.com/36895220/57559288-b8fcf000-7346-11e9-9648-8bf3756f67a9.png)

The Query only returns 7 rows in a simple table, and the total time is 30 seconds, :(

Now, I close OmniDB and open it again, and the result is different:

![image](https://user-images.githubusercontent.com/36895220/57559431-607a2280-7347-11e9-9d24-dcbf321c5df9.png)

I could not identify what the problem was

**Comment by wind39** [comment]
Hi @erdusa ,

Thanks for the use case. Did you execute anything before the query started to take 30 seconds? Did you cancel the previous query?

Regards,

William

**Comment by erdusa** [comment]
No, I didn't, I am thinking that the other possible slow query is not the problem

**Comment by erdusa** [comment]
Hi @wind39 
look at this new find

![image](https://user-images.githubusercontent.com/36895220/57571408-5c93e200-73d3-11e9-8c89-a3a1a6d9c95c.png)

It is the same query executed in both applications regardless of whether the query is cached or not

**Comment by rafaelthca** [comment]
Hi @erdusa,

So this specific query takes around 30 seconds every time you execute in OmniDB? When you click on Fetch All, does that return the rest of records quickly?

We need to try to reproduce this. Does this happen with other queries too?

**Comment by erdusa** [comment]
hi @rafaelthca 
Now I am running the query en omnidb and it takes 59 seconds every time, all time, and when I click on Fetch all takes 30 seconds, but in PGAdmin always takes 3.4 seconds

PgAdmin
![image](https://user-images.githubusercontent.com/36895220/57572727-58bc8b80-73e4-11e9-85d4-372e8c0af9c3.png)

OmniDB:
![image](https://user-images.githubusercontent.com/36895220/57572775-e4ceb300-73e4-11e9-9270-e16ef5120779.png)

click on fetch all:
![image](https://user-images.githubusercontent.com/36895220/57572781-04fe7200-73e5-11e9-86c1-5647ba6731bd.png)

And with others query also it happens, but in sometimes, I don't know which is the problem

**Comment by rafaelthca** [comment]
Ok thanks. Did that happen in 2.14.0 or earlier versions? If I create a custom build with additional debug messages, would you test for us? The idea is to install the custom build and reproduce the issue again. Then, send us the logs.

If you are willing to do that, please confirm the version. I can see you are using windows, app version, right?

**Comment by erdusa** [comment]
Hi @rafaelthca, Yes of course, I will do it.

I'm using the 2.15.0 OmniDB version, and yes it is Windows and app version

**Comment by wind39** [comment]
Hi @erdusa , @rafaelthca 

I stumped into this issue by chance, but I was executing the same very simple query against 2 OmniDB instances:

- Local app connected to a remote database via SSH tunnel;
- Remote server connected to a remote database, which was close to the server.

And the 30 seconds "penalty" happened only on the instance that was connected to the database via SSH tunnel, and not the remote one. This local app was open by several days already. Restarting the OmniDB app re-created the SSH tunnel and fixed the issue.

Perhaps you were connected via tunnel too?

Regards,

William
